pkg/state/state.go

```diff
@@ -619,7 +619,7 @@ func (st *HelmState) prepareSyncReleases(helm helmexec.Interface, additionalValu
 					flags = append(flags, "--wait-for-jobs")
 				}
 
-				if opts.ReuseValues || st.HelmDefaults.ReuseValues {
+				if !opts.ResetValues && (st.HelmDefaults.ReuseValues || opts.ReuseValues) {
 					flags = append(flags, "--reuse-values")
 				} else {
 					flags = append(flags, "--reset-values")
```

Comment:
we can reuse the logic. it can be a function.

Reply:
ok